Engine/src/Engine/ResourceManager.cpp

```diff
@@ -0,0 +1,164 @@
+	//Based on the provided filename, return the json file from a map, or load it from the system.
+	nlohmann::json ResourceManager::getJsonData(const std::string& Name)
+	{
+		//Initilaze the return value, and create a map iterator to check if the json is already loaded.
+		nlohmann::json data = nullptr;
+		std::unordered_map<std::string, nlohmann::json>::iterator it = m_jsons.find(Name);
+		
+		//If loaded, return the json. Else, load and store the json, then return it.
+		if (it != m_jsons.end())
+		{
+			//Get value based on map key "Name"
+			data = m_jsons.find(Name)->second;
+			
+			GE_CORE_WARN("[ResourceManager] Previously loaded " + Name + " found.");
+			return data;
+		}
+		
+		//Create an iterator to check if the file exists in the map (done since using m_filePaths[Name] will create a new entry if it doesn't exist).
+		std::unordered_map<std::string, std::string>::iterator jsonPath = m_filePaths.find(Name);
+			
+		//If json path is found (should be loaded into m_filePaths, currently upon initialization), load, store, then return the json.
+		if (jsonPath != m_filePaths.end())
+		{
+			std::string path = m_filePaths[Name];
+			std::string extension = path.substr(path.find_last_of('.') + 1);
+			
+			//Ensure the file is a json file
+			if (extension == "json")
+			{
+				GE_CORE_INFO("[ResourceManager] " + Name + " found.");
+				
+				data = nlohmann::json::parse(readSource(path));
+
+				m_jsons.insert(std::pair<std::string, nlohmann::json>(Name, data));
+				return data;
+			}
+		}
+
+		GE_CORE_INFO("[ResourceManager] " + Name + " not found.");
+		return nullptr;
+	}
+
+	//Based on the provided filename, return the shader source from a map, or load it from the system.
+	std::string ResourceManager::getShaderData(const std::string& Name)
+	{
+		//Initilaze the return value, and create a map iterator to check if the shader is already loaded.
+		std::string data;
+		std::unordered_map<std::string, std::string>::iterator it = m_shaders.find(Name);
+		
+		//If loaded, return the shader. Else, load and store the shader, then return it.
+		if (it != m_shaders.end())
+		{
+			//Get value based on map key "Name"
+			data = m_shaders.find(Name)->second;
+
+			GE_CORE_WARN("[ResourceManager] Previously loaded " + Name + " found.");
+			return data;
+		}
+		
+		//Create an iterator to check if the file exists in the map (done since using m_filePaths[Name] will create a new entry if it doesn't exist).
+		std::unordered_map<std::string, std::string>::iterator shaderPath = m_filePaths.find(Name);
+			
+		//If shader path is found (should be loaded into m_filePaths, currently upon initialization), load, store, then return the shader.
+		if (shaderPath != m_filePaths.end())
+		{
+			
+			std::string searchResult = m_filePaths[Name];
+			//Ensure the file is a shader file, based on the extension. 
+			std::string extension = searchResult.substr(searchResult.find_last_of('.') + 1);
+
+			if (extension == "fs" || extension == "vs")
```

Comment:
"fs", "vs", and "Json" are still magic variables. Should have these as private members.

Reply:
Ah, sorry I misunderstood what a magic variable was. This should be resolved